thetis/diagnostics.py

```diff
@@ -140,10 +154,36 @@ def __init__(self, field_2d, hessian_2d, gradient_2d=None, **kwargs):
         prob = LinearVariationalProblem(a, L, sol)
         kwargs.setdefault('solver_parameters', sp)
         self.solver = LinearVariationalSolver(prob, **kwargs)
+        self._isfrozen = True
 
     @PETSc.Log.EventDecorator("thetis.HessianRecoverer2D.solve")
     def solve(self):
         self.solver.solve()
         self.hessian_2d.assign(self._hessian)
         if self.gradient_2d is not None:
             self.gradient_2d.assign(self._gradient)
+
+
+class DynamicPressureCalculator(FrozenHasTraits):
```

Comment:
I would prefer to call this `KineticEnergyCalculator` as I think that's more commonly used term.

Reply:
Renamed as requested and added a base class.